meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register

```diff
@@ -100,31 +101,20 @@ class Pkcs11:
             cls._initialize()
         return cls()
 
-
-class SssCli:
-    @contextmanager
-    def i2c_session(*args, **kwds):
-        log.info("Connecting to ssscli...")
-        subprocess.check_call(["ssscli", "connect", "se05x", "t1oi2c", "none"])
-        try:
-            yield
-        finally:
-            log.info("Disconnecting from ssscli")
-            subprocess.check_call(["ssscli", "disconnect"])
-
+class FioCli:
     @classmethod
-    def read_id_list(cls) -> List[str]:
-        with cls.i2c_session():
-            out = subprocess.check_output(["ssscli", "se05x", "readidlist"])
-            lines = out.decode().splitlines()
-            # lines are like: Key-Id: 0Xf0000003   BINARY .....
-            return [x.split(" ")[1] for x in lines if x]
+    def read_id_list(self) -> List[str]:
+    command = ["fio-se05x-cli", "--list-objects", "all"]
+    process = subprocess.Popen(command, stdout=None, stderr=subprocess.PIPE)
+    stdout,stderr = process.communicate()
+    lines = stderr.decode().splitlines()
+    pattern = r'Key-Id:\s+(0x[0-9a-fA-F]+)'
```

Comment:
if a key-id always has the same length, a quantifier `{N}` has better performance than `+`

Reply:
the key id doesnt always have the same length as a string (it is a string representing any 32bit hex value)
reduced matches? um, I did try the above code with different lengths and it does as expected..what problem do you see with it?

Reply:
> the key id doesnt always have the same length as a string (it is a string representing any 32bit hex value) reduced matches? um, I did try the above code with different lengths and it does as expected..what problem do you see with it?

I remind sometimes ago Python re had problems with its NFA engine. (Like described here https://bytes.com/topic/python/answers/534624-how-get-longest-possible-match-pythons-module)

They may have already fixed it in newer versions. But I personally never trust 3rd-party re algos (not perl/libpcre ;-) and guard match groups on both sides. Like `\s+(0x[0-9a-fA-F]+)\b`